components/eventing-controller/pkg/backend/jetstreamv2/jetstream_internal_unit_test.go

```diff
@@ -0,0 +1,425 @@
+
+				}
+			},
+			wantErr: &backenderrors.FailedToSubscribeOnNATSError{OriginalError: nats.ErrJetStreamNotEnabled},
+		},
+		{
+			name:              "Update maxInFlight on the consumer side should be successful, when the subscription config changes",
+			givenSubscription: subWithOneType,
+			givenSubscriptionMap: map[SubscriptionSubjectIdentifier]Subscriber{
+				jsSubKey: subscriberMock,
+			},
+			givenMocks: func(sub *v1alpha2.Subscription, jsBackend *JetStream, jsCtx *mocks.JetStreamContext, jsSubKey SubscriptionSubjectIdentifier) {
+				for _, eventType := range sub.Status.Types {
+					jsSubject := jsBackend.getJetStreamSubject(sub.Spec.Source, eventType.CleanType, sub.Spec.TypeMatching)
+					jsSubKey := NewSubscriptionSubjectIdentifier(sub, jsSubject)
+					subscriberMock.On("IsValid").Return(true)
+
+					subWithOneType.Spec.Config[v1alpha2.MaxInFlightMessages] = "20"
+					jsCtx.On("ConsumerInfo", jsBackend.Config.JSStreamName, jsSubKey.ConsumerName()).Return(defaultConsumerInfo, nil)
+					jsCtx.On("UpdateConsumer", jsBackend.Config.JSStreamName, mock.AnythingOfType("*nats.ConsumerConfig")).Return(defaultConsumerInfo, nil)
+				}
+			},
+			givenAssertions: func(sub *v1alpha2.Subscription, jsBackend *JetStream, jsCtx *mocks.JetStreamContext, jsSubKey SubscriptionSubjectIdentifier) {
+				// check if the expected methods got called
+				for _, eventType := range sub.Status.Types {
+					jsSubject := jsBackend.getJetStreamSubject(sub.Spec.Source, eventType.CleanType, sub.Spec.TypeMatching)
+					jsSubKey := NewSubscriptionSubjectIdentifier(sub, jsSubject)
+
+					jsCtx.AssertCalled(t, "ConsumerInfo", jsBackend.Config.JSStreamName, jsSubKey.ConsumerName())
+
+					defaultConsumerInfo.Config.MaxAckPending = 20
+					jsCtx.AssertCalled(t, "UpdateConsumer", jsBackend.Config.JSStreamName, &defaultConsumerInfo.Config)
+
+					jsCtx.AssertNotCalled(t, "AddConsumer", jsBackend.Config.JSStreamName, mock.AnythingOfType("*nats.ConsumerConfig"))
+				}
+			},
+		},
+	}
+	for _, testCase := range testCases {
+		tc := testCase
+		t.Run(tc.name, func(t *testing.T) {
+			// given
+			testEnv := getTestEnvironmentWithMock(t)
+			if testCase.givenSubscriptionMap != nil {
+				testEnv.jsBackend.subscriptions = testCase.givenSubscriptionMap
+			}
+			for _, eventType := range testCase.givenSubscription.Status.Types {
+				jsSubject := testEnv.jsBackend.getJetStreamSubject(testCase.givenSubscription.Spec.Source, eventType.CleanType, testCase.givenSubscription.Spec.TypeMatching)
+				jsSubKey := NewSubscriptionSubjectIdentifier(testCase.givenSubscription, jsSubject)
+
+				// mock the expected calls
+				tc.givenMocks(testCase.givenSubscription, testEnv.jsBackend, testEnv.jsCtxMock, jsSubKey)
+			}
+
+			// when
+			err := testEnv.jsBackend.syncNATSConsumers(testCase.givenSubscription, callback)
+
+			// then
+			// check if error is expected
+			if testCase.wantErr == nil {
+				require.NoError(t, err)
+			} else {
+				require.Error(t, err)
```

Comment:
This call is not required as there is require.ErrorIs() in the next line

Reply:
done